### PR TITLE
Add troubleshooting for packaging

### PR DESCRIPTION
**Pull request by davidjharder**
Resolves https://github.com/getsolus/help-center-docs/issues/411

### CODE REVIEW

**Comment by davidjharder** [comment]
Draft until I go through all the items I tossed on the accompanying issue

**Comment by davidjharder** [comment]
Added a mention of `go-task solbuild-reset`, part of https://github.com/getsolus/help-center-docs/issues/521

File: docs/packaging/troubleshooting-packaging.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+If you interupt a build sometimes `solbuild` does not unmount cleanly, and you will need to unmount manually:
+
+```bash
+sudo umount -f /var/lib/solbuild/roots/unstable-x86_64 
+```
+
+### Solbuild is eating up all my disk space
+
+The `solbuld` cache at `/var/cache/solbuild` and can grow to tens of gigabytes easily. Recover that diskspace by running the "delete cache (`dc`) command":
```
Comment (davidjharder) [inline docs/packaging/troubleshooting-packaging.md:64]:
```suggestion
The `solbuld` cache at `/var/cache/solbuild` and can grow to tens of gigabytes easily. Recover that disk space by running the "delete cache (`dc`) command":
```

File: docs/packaging/troubleshooting-packaging.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+If you interupt a build sometimes `solbuild` does not unmount cleanly, and you will need to unmount manually:
+
+```bash
+sudo umount -f /var/lib/solbuild/roots/unstable-x86_64 
+```
+
+### Solbuild is eating up all my disk space
+
+The `solbuld` cache at `/var/cache/solbuild` and can grow to tens of gigabytes easily. Recover that disk space by running the "delete cache (`dc`) command":
```
Comment (davidjharder) [inline docs/packaging/troubleshooting-packaging.md:64]:
```suggestion
The `solbuild` cache at `/var/cache/solbuild` and can grow to tens of gigabytes easily. Recover that disk space by running the "delete cache (`dc`) command":
```

File: docs/packaging/troubleshooting-packaging.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+If your build fails immediately this  message, and your are using a **git source**, you should clear the `solbuild` git sources cache:
+
+```bash
+sudo rm -rf /var/lib/solbuild/sources
+```
+
+### "mount: /var/lib/solbuild/roots/unstable-x86_64: WARNING: source write-protected, mounted read-only"
+
+If you interupt a build sometimes `solbuild` does not unmount cleanly, and you will need to unmount manually:
```
Comment (davidjharder) [inline docs/packaging/troubleshooting-packaging.md:56]:
```suggestion
If you interrupt a build sometimes `solbuild` does not unmount cleanly, and you will need to un-mount manually:
```

File: docs/packaging/troubleshooting-packaging.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+The `solbuild` cache at `/var/cache/solbuild` and can grow to tens of gigabytes easily. Recover that disk space by running the "delete cache (`dc`) command":
+
+```bash
+sudo solbuild dc
+```
+
+## An old package using `%configure` doesn't build
+
+Replace `%configure` with `%reconfigure` and try again.
+If that doesn't work, try replacing `%configure` with `./configure $(sed 's|--runstatedir=/run||g' <<< "%CONFOPTS%")`
```
Comment (Staudey) [inline docs/packaging/troubleshooting-packaging.md:73]: Maybe make it clear that if the error is explicitly `configure: error: unrecognized option: '--runstatedir=/run'` **then** one should use this workaround first.

Also there now exists a macro to replace the lengthy `sed` instruction: `%configure_no_runstatedir` (https://github.com/getsolus/ypkg/blob/master/ypkg2/rc.yml#L5)

Comment (davidjharder) [reply docs/packaging/troubleshooting-packaging.md:73]: oh cool

File: docs/packaging/troubleshooting-packaging.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+cat package.yml | gh gist create --public
+```
+
+Send the output of `solbuild` to a public Gist:
+
+```bash
+go-task &| gh gist create --public
+```
+
+*Note: the usual `solbuild`` output will not be shown when this command is running, and it will take at least a few seconds*
```
Comment (Staudey) [inline docs/packaging/troubleshooting-packaging.md:28]:
```suggestion
*Note: the usual `solbuild` output will not be shown when this command is running, and it will take at least a few seconds*
```

File: docs/packaging/troubleshooting-packaging.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+summary: Common packaging issues and how to fix them
+---
+
+# Troubleshooting for Packagers
+
+This page covers common packaging issues and how to fix them. We also show you how to share your packaging work so you can ask for help in the Solus Matrix rooms.
+
+## Asking for help
+
+When you run into a problem that you don't know how to fix you should ask for help in the [Solus Packaging matrix room](/user/contributing/getting-involved.md#matrix-chat). But first you should send the `package.yml` file and `solbuild` output to somewhere others can see it; we recommend you use [GitHub Gists](https://gist.github.com/). These commands assume you have set up `github-cli`.
```
Comment (Staudey) [inline docs/packaging/troubleshooting-packaging.md:12]:
```suggestion
When you run into a problem that you don't know how to fix you should ask for help in the [Solus Packaging Matrix room](/user/contributing/getting-involved.md#matrix-chat). But first you should send the `package.yml` file and `solbuild` output to somewhere others can see it; we recommend you use [GitHub Gists](https://gist.github.com/). These commands assume you have set up `github-cli`.
```

File: docs/packaging/troubleshooting-packaging.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+Your `source` tarball does not match the checksum next to it in `package.yml`. You should  use `go-task new` or `go-task update` to write the correct checksum automatically.
+
+###  ffmpeg-chromium won't compile
+
+Make Harvey do it instead
+
+### "Failed to build packages err="Failed to fetch source https://github.com/project/blah.git#{commit ref here}, reason: exit status 1\n"
+
+If your build fails immediately this  message, and your are using a **git source**, you should clear the `solbuild` git sources cache:
```
Comment (Staudey) [inline docs/packaging/troubleshooting-packaging.md:48]:
```suggestion
If your build fails immediately this  message, and you are using a **git source**, you should clear the `solbuild` git sources cache:
```

File: docs/packaging/troubleshooting-packaging.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+```bash
+sudo rm -rf /var/lib/solbuild/sources
```
Comment (Staudey) [inline docs/packaging/troubleshooting-packaging.md:51]:
```suggestion
sudo rm -rf /var/lib/solbuild/sources/git
```

Otherwise it would be the whole sources cache, not just git. Which would be fine too I guess, but then I would rephrase the last part of the sentence above to just say "sources cache"

File: docs/packaging/troubleshooting-packaging.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+If you interrupt a build sometimes `solbuild` does not unmount cleanly, and you will need to unmount manually:
+
+```bash
+sudo umount -f /var/lib/solbuild/roots/unstable-x86_64 
+```
+
+### Solbuild is eating up all my disk space
+
+The `solbuild` cache at `/var/cache/solbuild` and can grow to tens of gigabytes easily. Recover that disk space by running the "delete cache (`dc`) command":
```
Comment (Staudey) [inline docs/packaging/troubleshooting-packaging.md:64]:
```suggestion
The `solbuild` cache at `/var/cache/solbuild` can grow to tens of gigabytes easily. Recover that disk space by running the "delete cache (`dc`)" command:
```
Either remove the "and", or change to "is at" or "resides at" (or similar)

**Review by Staudey — COMMENTED** [review]
I'm not sure if you're planning to leave the Harvey `ffmpeg-chromium` joke in or not, but I'm okay with it either way :grin:

**Comment by davidjharder** [comment]
> I'm not sure if you're planning to leave the Harvey `ffmpeg-chromium` joke in or not, but I'm okay with it either way 😁

I'll remove it. We have lots of other places to make stupid jokes
